be/src/star_cache/eviction_policy.h

```diff
@@ -0,0 +1,37 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include <atomic>
+#include "common/status.h"
+
+namespace starrocks {
+
+template <typename T>
+class EvictionPolicy {
+public:
+    virtual ~EvictionPolicy() = default;
+
+	// Add the given id to the evict component
+	virtual bool add(const T& id) = 0;
+
+	// Record the hit of the id
+	virtual bool touch(const T& id) = 0;
+
+	// Evict some items in current component
+	virtual void evict(size_t count, std::vector<T>* evicted) = 0;
+
+	// Evict some items in current component to store given `id`
+    // This function is useful when evicting some items which are devided into
```

Comment:
divide?

and what's `count` for?

`evicted` is for holding evicted item?

Reply:
> divide?
> 
> and what's `count` for?
> 
> `evicted` is for holding evicted item?

`count` is used to evict multiple elements in one evict, act as a `batch_evict`. 
`evicted` holds the evicted items.